### ISSUE TITLE
iteratively appending data to a pkl file

### ISSUE THREAD

**Issue by ferreirafabio** [episodeyang/ml_logger#1]
Hi,

apparently, iteratively appending data to a .pkl file with 

`logger.log_data(file_name, data=new_data)`

doesn't seem to work in my implementations. When executing `logger.load_pkl(file_name)`, I always get the first element that I inserted and the rest is missing. I've checked the implementation and the byte append flags (in server.py) are 'ab' which seem correctly set. I face this issue in sequential and parallel mode. Have you tested your `log_data` function somewhere since I can't see a test for it in the repository?

P.S.: `logger.flush(file_name)` unfortunately didn't solve the issue.

Thanks in advance!

**Comment by episodeyang** [comment]
@ferreirafabio @jonasrothfuss Thanks for reporting the issue here! 

I'm reviewing the usage of the `load_pickle` method in the current `ml_logger` code base. It mostly sounds right that we should return the entire file, I just want to double check to make sure that it is reasonable for people to expect doing this even for a single pickle object in a file container:

```python
weights, = logger.load_pkl('modules/discriminator.pkl')
# Notice the comma on the top
```

then for all of the weights, you get:

```python
versions_of_weights = logger.load_pkl('modules/discriminator.pkl')
print(version_of_weights)
# [
#    {w1: np.array(..., b1: np.array(... }, 
#    {w1: np.array(..., b1: np.array(... }, 
#  ]
```

**Comment by episodeyang** [comment]
I'm looking at this line and am wondering why this is doing this:

https://github.com/episodeyang/ml_logger/blob/dd8487640966ecbaa15e3f35b1f1e9d452659ab4/tests/test_ml_logger.py#L111

**Comment by episodeyang** [comment]
lol I found the error: If you squint the line below at the end of the line, there is a coma that shouldn't be there:

https://github.com/episodeyang/ml_logger/blob/e658f17a8786c1e7d04b29730538b2da3bd0a158/ml_logger/ml_logger.py#L452

**Comment by episodeyang** [comment]
@ferreirafabio @jonasrothfuss Now addressed in #2

**Comment by episodeyang** [comment]
@ferreirafabio PR merged, published as `v0.1.14` on PyPI: https://pypi.org/project/ml-logger/

**Comment by ferreirafabio** [comment]
It work's like a charm. Thanks a lot for the fast response and reaction.
